index.rst

```diff
@@ -1,42 +1,621 @@
+
+The solution we have adopted for PSF models has been referred as both :ref:`CoaddPsf <coaddpsf>` (from :py:class:`lsst.meas.algorithms.CoaddPsf`) and :ref:`StackFit <stackfit>` (after the shear estimation technique where it was developed), and is essentially a form of lazy evaluation.  When a PSF model image is requested at a point, we simply evaluate the PSF models for all of the input images at that point, transform them to the correct coordinate system, and compute the weighted sum on demand.  We typically assume that the PSF is constant over the scale of a single astronomical object, and hence this reduces the number of PSF model evaluations from the number of pixels to the number of detected objects.  When an object lies on a boundary or a region with masked pixels, the true PSF is discontinuous and the constant-PSF assumption is not valid.  At present, we simply flag objects for which this is true, but this may not work when the number of input images is large; in this regime the number of border and masked regions increases, though the severity of the discontinuities decreases as well.
+
+Our current approach for coadd uncertainty propagation is to compute and store only the variance.  We will likely expand this in the future to storing some approximation to the covariance (e.g. by modeling it as constant within regions where the number of input images is constant).
+
+Direct coadds are lossy, requiring some trade-off between image quality (PSF size) and depth (SNR).  This can be easily seen from :eq:`eq:coaddpsf`: including an image with a PSF larger than the current weighted mean PSF always increases the size of the final PSF, regardless of the depth of the new image.
+
+
+PSF-Matched Coadds
+------------------
+
+In PSF-matched coaddition, input images are convolved by a kernel that matches their PSF to a predefined constant PSF before they are combined.  If :math:`\phi_\mathrm{pm}({\bf r})` is the predefined PSF for the coadd, then the matching kernel :math:`K_i({\bf r}, {\bf s})` is defined such that
+
+.. math::
+  \sum_{\bf u} \! K_i({\bf r}, {\bf u}) \, \phi_i({\bf u}, {\bf s})
+    = \phi_\mathrm{pm}({\bf r}-{\bf s})
+
+Typically :math:`K` is parametrized as a smoothly varying linear combination of basis functions.  The details of fitting it given a target coadd PSF and input image PSF models is beyond the scope of this document.
+
+Because deconvolution is (at best) noisy, convolution with :math:`K_i` will generally increase the size of the PSF.  This highlights the big disadvantage of PSF-matched coadds: the images with the best seeing must be degraded to match a target PSF whose sizes is determined by the worst of the images to be included in the coadd.  Thus PSF-matched coadds must either include only the best-seeing images (sacrificing depth) or suffer from a worst-case coadd PSF.
+
+After PSF-matching, the coadd is constructed in the same way as a direct coadd:
```

Comment:
Is w chosen as outlined in Direct Coadds? Since all the images have been convolved to have the same PSF, do we weight using the area of the avaerage original PSF of the un-convolved image as before, or do we just not use the PSF size and weight just based on the noise?